src/transformers/generation_utils.py

```diff
@@ -712,15 +713,33 @@ def _get_logits_processor(
             processors.append(ForcedEOSTokenLogitsProcessor(max_length, forced_eos_token_id))
         if remove_invalid_values is True:
             processors.append(InfNanRemoveLogitsProcessor())
+        processors = self._merge_criteria_processor_list(processors, logits_processor)
         return processors
 
-    def _get_stopping_criteria(self, max_length: Optional[int], max_time: Optional[float]) -> StoppingCriteriaList:
-        stopping_criteria = StoppingCriteriaList()
+    def _get_stopping_criteria(
+        self, max_length: Optional[int], max_time: Optional[float], stopping_criteria: Optional[StoppingCriteriaList]
+    ) -> StoppingCriteriaList:
+        criteria = StoppingCriteriaList()
         if max_length is not None:
-            stopping_criteria.append(MaxLengthCriteria(max_length=max_length))
+            criteria.append(MaxLengthCriteria(max_length=max_length))
         if max_time is not None:
-            stopping_criteria.append(MaxTimeCriteria(max_time=max_time))
-        return stopping_criteria
+            criteria.append(MaxTimeCriteria(max_time=max_time))
+        criteria = self._merge_criteria_processor_list(criteria, stopping_criteria)
+        return criteria
+
+    def _merge_criteria_processor_list(
+        self, default_list: StoppingCriteriaList, custom_list: StoppingCriteriaList
+    ) -> StoppingCriteriaList:
+        if custom_list is None:
+            return default_list
+        for default in default_list:
+            for custom in custom_list:
+                if type(custom) is type(default):
+                    raise ValueError(
+                        f"A custom stopping criteria or logits processor of type {type(custom)} was passed to `generate` which is already created with an argument or model config."
```

Comment:
```suggestion
                        f"A custom stopping criteria or logits processor of type {type(custom)} was passed "
                        "to `generate` which is already created with an argument or model config."
```

Reply:
Also, since this error won't be easy to solve for the user can we try to give her/him as much guidance as possible? we could a) Figure out if it's a `stopping criteria` or a `logits processor` from the type. And then b) IMO we should display as much information as possible. E.g.
```
object_type = "stopping criteria" if isinstance(custom, StoppingCriteria) else "logits processor"
f"A custom {object_type} of type {type(custom)} with values {custom} has been passed to `generate`, but it has already been created with the values {default}. {default} has been created by passing the corresponding arguments to generate or by the config's default values."

Reply:
Something like this maybe, there is probably a better way to phrase it

Reply:
Actually what we could do in a second step is to enforce that the __init__ names of the processors have to correspond 1-to-1 to the inputs of `generate()` and then we can actually pass the arguments that need to set to `None` in generate to allow this. Maybe a bit overkill in a first step, but we could do this if we encounter issues about it

Reply:
I refined the error message to include a little bit more detail. Let me know if you think it is clearer now.